domain/application/state/application.go

```diff
@@ -875,33 +871,160 @@ INSERT INTO k8s_service (*) VALUES ($cloudService.*)
 			return errors.Errorf(
 				"querying cloud service for application %q and provider id %q: %w", applicationName, providerID, err)
 		}
-		if err == nil {
-			return nil
+		if errors.Is(err, sqlair.ErrNoRows) {
```

Comment:
Instead of separate `insertCloudServiceNetNode` and `insertCloudServiceDevice` methods, the whole block of code could be implemented in a single method. The single method could call the above or just roll the methods together - there's not a lot of code and avoids returning and then passing the net node uuid.